spec/helpers.lua

```diff
@@ -892,6 +892,33 @@ local function wait_for_invalidation(key, timeout)
   end, timeout)
 end
 
+--- Waits for the termination of a pid.
+-- @param pid_path Filename of the pid file.
+-- @param timeout (optional) in seconds, defaults to 10.
+local function wait_pid(pid_path, timeout)
+  local pid
+  local fd = io.open(pid_path)
+  if fd then
+    pid = fd:read("*l")
+    fd:close()
+  end
+
+  if pid then
+    local max_time = ngx.now() + (timeout or 10)
+    repeat
+      if pl_utils.execute("ps -p " .. pid .. " >/dev/null 2>&1") then
+        ngx.sleep(0.05)
+      else
+        return
```

Comment:
It is somewhat emphasized in the contributing file that returning early is an appreciated practice. Not a blocker here but I do think it is a good practice to build and enforce:

```lua
if not pl_utils.execute("ps -p " .. pid .. " >/dev/null 2>&1") then
  return
end

-- still running, let's wait some more
ngx.sleep(0.05)
```